src/include/Token.h

```diff
@@ -33,54 +33,64 @@ enum struct TokenType {
     EOL, END, Error
 };
 
+inline std::string tokTypeToString(const TokenType type) {
+    switch (type) {
+        case TokenType::Comma: return ",";
+        case TokenType::LeftBrace: return "{";
+        case TokenType::RightBrace: return "}";
+        case TokenType::LeftParen: return "(";
+        case TokenType::RightParen: return ")";
+        case TokenType::And: return "and";
+        case TokenType::Or: return "or";
+        case TokenType::Ampersand: return "&";
+        case TokenType::Plus: return "+";
+        case TokenType::Minus: return "-";
+        case TokenType::Star: return "*";
+        case TokenType::Slash: return "/";
+        case TokenType::BangEqual: return "!=";
+        case TokenType::EqualEqual: return "==";
+        case TokenType::Greater: return ">";
+        case TokenType::GreaterEqual: return ">=";
+        case TokenType::Lesser: return "<";
+        case TokenType::LesserEqual: return "<=";
+        case TokenType::True: return "True";
+        case TokenType::False: return "False";
+        case TokenType::Def: return "def";
+        case TokenType::Let: return "let";
+        case TokenType::Equal: return "=";
+        case TokenType::Bang: return "!";
+        case TokenType::If: return "if";
+        case TokenType::Else: return "else";
+        case TokenType::Elif: return "elif";
+        case TokenType::While: return "while";
+        case TokenType::Return: return "return";
+        case TokenType::Next: return "next";
+        case TokenType::Break: return "break";
+        case TokenType::Print: return "print";
+        case TokenType::Scan: return "scan";
+        case TokenType::Err: return "err";
+        case TokenType::EOL: return "EOL";
+        case TokenType::END: return "EOF";
+        case TokenType::String:
```

Comment:
c++ switch cases fall through, right? so in this case `String` and `Integer` would both print `"Literal"`?

Reply:
Yeah, I wanted to add the fall through attribute to make that explicit, but apparently it’s C++17. I’m not sure it’s worth printing seperate things for the different literal types, since it should be obvious from the value what they are; e.g. in the pretty printer it’ll print “(literal “hi”)”, and I’m not sure it adds much to change “literal” to, say, “string literal”.